### DISCUSSION TITLE
Check the "See all buying options page" for Amazon

### DISCUSSION THREAD

**Discussion by B1022** [jef/streetmerchant#803] [Ideas and feature requests]

**Comment by us1v1an** [comment]
I agree, there are many listings on Amazon that have no buybox on the main page. The only way to get accurate info is to check the "all offers" page. 

For example:

https://www.amazon.com/dp/B08L8JNTXQ

https://www.amazon.com/gp/offer-listing/https://www.amazon.com/dp/B08L8JNTXQ

**Comment by jef** [comment]
I'm sure we could do something like this, but I personally don't want to go down into other links and scrape those pages before solving some other enhancements. I think this is fine, but low on my personal list. If anyone else wants to take a crack at it, feel free!

**Comment by jef** [comment]
We'd probably add another label to look for. Like `external` or something.

**Comment by us1v1an** [comment]
Ideally, it should look for the listing that's sold by Amazon, or any listing below a certain price. But for the purpose of this tool, the only listing that really matters is the one sold by Amazon. Everyone else is a reseller and their price will always be much higher than retail price.

**Comment by J-Hinton** [comment]
Personally, I feel we should be only using the Buy Box. This is designed for a reason... as a seller you must reach criteria to win the buy box (e.g. trusted seller, competitive price, fba or prime self-fulfilled, new items only, in stock etc.). Any seller meeting their hidden criteria is able to win the buy box.
If we go down the route of using all sellers, you're going to get a load of notifications of in stock for resellers charging extremely high prices.

I'm happy to have a new environment variable to specify the type of Amazon seller; Buy-box (default) or all. But I don't think we should be forcing everyone down to having all sellers of a product.

**Comment by us1v1an** [comment]
I agree with you that we should only look for offers sold by Amazon. However, there are many times when something comes in stock (offer by Amazon) and gets sold out before the buybox even gets activated. 

Therefore, I think the best option is to make SM search for offers by Amazon only in the “all offers” page. Searching only the main page for the buybox is never going to work, specially for these high in demand products that come in stock and get sold out in seconds.

**Comment by J-Hinton** [comment]
@B1022 Please can you send an example link. I don't see any products with the class *olpBuyColumn*

**Comment by aesth0x** [comment]
Amazon sets rate limit on the /offer-listing page quite harshly. I tried and started to see Captcha very soon.

**Comment by HawkmoonDC** [comment]
Where do I enter the code? I have a listing that lost it's buy box feature (even though I am still the only seller)
